packages/e2e/detox/integration/common/authenticator.ts

```diff
@@ -22,24 +25,55 @@ const getCountryCode = (status: string) => {
 When(
   'I type my {string} with status {string}',
   async (loginMechanism: string, status: string) => {
-    let text = '';
+    let text = '',
+      usernameAttribute = loginMechanism.toUpperCase();
     if (loginMechanism === 'email') {
       text = `${getUserAlias(status)}@${process.env.DOMAIN}`;
     } else if (loginMechanism === 'username') {
       text = getUserAlias(status);
     } else if (loginMechanism === 'phone number') {
       text = `${getCountryCode(status)}${process.env.PHONE_NUMBER}`;
+      usernameAttribute = 'Phone Number';
     }
-    await element(by.id(`authenticator__text-field__input-username`)).typeText(
-      text
-    );
+
+    const inputField = by
+      .type('UITextField')
+      .withDescendant(by.label(`Enter your ${usernameAttribute}`));
+    await element(inputField).typeText(text);
```

Comment:
I added a test id for the formfield inputs, however on the SignIn screen, the input ID is always `authenticator__text-field__input-username` regardless of what the label says, so had to get a little creative here.